hashintel/hash
# H-583, H-665: Improve settings sidebar UX, make type editor alert modals consistent

## 🌟 What is the purpose of this PR?

- H-583: Update the settings sidebar
  - so that only one menu at a given depth can be expanded
  - to remove the toggle caret at the top level
  - to make items progressively closer together as you go down the tree
- H-665: Make the alert modals in the type editor closer to the design, and consistent (the 'convert to link style' modal was different to the modals you get when trying to add chosen parents)

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:



### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ❓ How to test this?



1. Check out locally or view the deployment (frontend changes only)
2. Click around the settings sidebar, try behavior on refreshing at different depths
3. In the type editor, try convert to link type, or adding a link type to a non-link type

## 📹 Demo



#### Settings sidebar
https://github.com/hashintel/hash/assets/37743469/e9a378e0-73db-413e-8f11-f3df272a59e7

#### Modals in type editor

![Screenshot 2023-09-22 at 11 35 33](https://github.com/hashintel/hash/assets/37743469/0f984bab-4b2d-43b9-9ef2-d5567a7ad08b)


![Screenshot 2023-09-22 at 11 27 14](https://github.com/hashintel/hash/assets/37743469/0f78ee0e-b7a5-417f-9f96-4271145d424d)
